tests/Unit/ParallelAlgorithms/Amr/Policies/Test_Policies.cpp

```diff
@@ -10,31 +10,54 @@
 #include "Framework/TestHelpers.hpp"
 #include "ParallelAlgorithms/Amr/Policies/Isotropy.hpp"
 #include "ParallelAlgorithms/Amr/Policies/Policies.hpp"
+#include "ParallelAlgorithms/Amr/Policies/RefinementLimits.hpp"
 
 namespace {
 void test_equality() {
   INFO("Equality");
-  CHECK(amr::Policies{amr::Isotropy::Anisotropic} ==
-        amr::Policies{amr::Isotropy::Anisotropic});
-  CHECK(amr::Policies{amr::Isotropy::Anisotropic} !=
-        amr::Policies{amr::Isotropy::Isotropic});
+  CHECK(amr::Policies{amr::Isotropy::Anisotropic, amr::RefinementLimits{}} ==
+        amr::Policies{amr::Isotropy::Anisotropic, amr::RefinementLimits{}});
+  CHECK(amr::Policies{amr::Isotropy::Anisotropic, amr::RefinementLimits{}} !=
+        amr::Policies{amr::Isotropy::Isotropic, amr::RefinementLimits{}});
 }
 
 void test_pup() {
   INFO("Serialization");
-  test_serialization(amr::Policies{amr::Isotropy::Anisotropic});
+  test_serialization(
+      amr::Policies{amr::Isotropy::Anisotropic, amr::RefinementLimits{}});
 }
 
 void test_option_parsing() {
   INFO("Option Parsing creation");
   const auto isotropies =
       std::array{amr::Isotropy::Anisotropic, amr::Isotropy::Isotropic};
+  const auto default_limits = std::make_optional<amr::RefinementLimits>();
+  const auto specified_limits =
+      std::make_optional<amr::RefinementLimits>(1, 5, 3, 7);
   for (const auto isotropy : isotropies) {
-    std::ostringstream creation_string;
-    creation_string << "Isotropy: " << isotropy;
-    const auto policies =
-        TestHelpers::test_creation<amr::Policies>(creation_string.str());
-    CHECK(policies == amr::Policies{isotropy});
+    CAPTURE(isotropy);
+    {
+      INFO("specified limits");
+      std::ostringstream creation_string;
+      creation_string << "Isotropy: " << isotropy << "\n";
+      creation_string << "RefinementLimits:\n"
+                      << "  MinimumRefinementLevel: 1\n"
+                      << "  MaximumRefinementLevel: 5\n"
+                      << "  MinimumResolution: 3\n"
+                      << "  MaximumResolution: 7\n";
```

Comment:
I think you can reduce the verbosity of these options to:

```
Limits:
  RefinementLevel: [0, 8]  # or Auto
  NumGridPoints: [2, 12]  # or Auto
```

(parse the two options as `Options::Auto<std::pair>` and add some parse errors in the constructor)

This accomplishes the following:

- The name `RefinementLimits` is redundant with `Min/MaxRefinementLevel`, but doesn't fit `Min/MaxResolution`, so better name it just `Limits` (that's clear enough in the AMR context).
- The name `Resolution` doesn't fit together with the domain creators, which specify `InitialGridPoints`. To make it clear that the two options refer to the same thing, name it `NumGridPoints`. (I understand that p-refinement is called "resolution" in some places and I don't have better naming for that at the moment, but I think making this option consistent with the domain creator is important.)
- I find the `Minimum` and `Maximum` naming overly verbose, and a simple pair of numbers quite clear and easier to read.
- This is also short enough that you don't need `Auto` for the parent option (`Limits`), but just for each of the two pairs. So I'd avoid hiding the two options behind a `Limits: Auto`. Instead you can set `Auto` for either h-refinement, or p-refinement, or both.